app/controllers/authz_controller.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class AuthzController < ApplicationController
+  before_action :authenticate_account!, unless: :public_route?
+  skip_before_action :verify_authenticity_token
+
+  PUBLIC_PATHS = [
+    'diag',
+    'ambassador',
+    'accounts/sign_in',
+    'accounts/sign_out',
+    'accounts/sign_up',
+    'health/alive',
+    'health/ready'
+  ]
+
+  def verify
+    return head(:ok) if public_route?
+
+    response.set_header('Authorization', "Bearer #{create_token}")
+    head :ok
+  end
+
+  private
+
+  def create_token
```

Comment:
Could you use a non Doorkeeper method since we are going to delete this?
I suggest using gem peatio